backend/src/main/java/com/google/sps/servlets/SearchMatchServlet.java

```diff
@@ -142,9 +152,33 @@ private void sendMatchResponse(HttpServletResponse response, Match match) throws
     matchExists.put(JSON_FIRST_PARTICIPANT_USERNAME, match.getFirstParticipantUsername());
     matchExists.put(JSON_SECOND_PARTICIPANT_USERNAME, match.getSecondParticipantUsername());
     matchExists.put(JSON_DURATION, match.getDuration());
-
+    try {
+      emailNotifier = ensureEmailNotifier();
+    } catch (GeneralSecurityException e) {
+      e.printStackTrace();
+    }
+    try {
+      emailNotifier.sendMatchEmail(getUsername(), userService.getCurrentUser().getEmail());
+    } catch (MessagingException e) {
+      e.printStackTrace();
+    }
     // Send the JSON back as the response
     response.setContentType("application/json");
     response.getWriter().println(matchExists.toString());
   }
+
+  /** Retrieve user email address via Users API and parse for username */
+  private String getUsername() {
+    UserService userService = UserServiceFactory.getUserService();
+    String email = userService.getCurrentUser().getEmail();
+    return email != null ? email.split("@")[0] : null;
+  }
+
+  private EmailNotifier ensureEmailNotifier() throws GeneralSecurityException, IOException {
+    if (this.emailNotifier == null) return null;
```

Comment:
Is this logic right? I think we want to create a new emailNotifier when it's null and return it when it's not?

Reply:
yeah it is sorry i was trying to copy the logic from a ternary operator standpoint